catalog/resource_sql_table.go

```diff
@@ -431,6 +467,24 @@ func columnChangesCustomizeDiff(d *schema.ResourceDiff) error {
 					return fmt.Errorf("changing the 'type' of an existing column is not supported")
 				}
 			}
+		} else {
+			oldColsNameToMap := make(map[string]map[string]interface{})
```

Comment:
Ditto: this is really nested. Rule of thumb is 2 levels of indentation, with a 3rd allowed for `if err != nil {}` checks because Golang. More than that means that the function complexity is probably too high, or it is difficult to review.